osquery/dispatcher/scheduler.cpp

```diff
@@ -211,7 +211,7 @@ void SchedulerRunner::start() {
 
     // GLog is not re-entrant, so logs must be flushed in a dedicated thread.
     if ((i % 3) == 0) {
-      relayStatusLogs(true);
```

Comment:
This is called on the main scheduler thread (it actually should be moved to and independent thread), so making `relayStatusLogs` synchronous has the potential of delaying this thread.

Reply:
Nice catch.  I blindly changed it to fit the new interface. This function does not need to be called here at all, removing it.